dynomitemanager/src/main/java/com/netflix/dynomitemanager/defaultimpl/DynomitemanagerConfiguration.java

```diff
@@ -704,11 +713,23 @@ public String getCommaSeparatedCassandraHostNames() {
     public boolean isEurekaHostSupplierEnabled() {
 	return configSource.get(CONFIG_IS_EUREKA_HOST_SUPPLIER_ENABLED, DEFAULT_IS_EUREKA_HOST_SUPPLIER_ENABLED);
     }
-    
+
     // Redis compatibility
     @Override
     public String getRedisCompatibleEngine() {
 	return configSource.get(CONFIG_DYNO_REDIS_COMPATIBLE_ENGINE, DEFAULT_REDIS_COMPATIBLE_ENGINE);
     }
-    
+
+    // Redis
+    // =====
+
+    /**
+    * Get the full path to the redis.conf configuration file.
+    * @return the {@link String} full path to the redis.conf configuration file
+    */
+    @Override
+    public String getRedisConf() {
```

Comment:
I have spent more time looking into this.

I don't think it's a good idea to use `IConfigSource` in `RedisStorageProxy`. 

### My thoughts on the appropriate solution

IMO, it's better to keep the configuration properties code centralized in `DynomitemanagerConfiguration`. I can organize the code, variable/constant names, etc. so that the file (while long) will still be easy to maintain.

The reason I don't think we should put `IConfigSource` in `RedisStorageProxy` follows:

- In `InjectedWebListener` line 85, the `initialize()` method of `IConfiguration` is called:
- In `InjectedWebListener` line 121, `IConfiguration` is bound to `DynomitemanagerConfiguration`.
- The `DynomitemanagerConfiguration.initialize()` method calls 3 methods:

```java
setupEnvVars();
// This is IConfigSource.initialize()!
this.configSource.initialize(ASG_NAME, REGION);
setDefaultRACList(REGION);
```

**The problem**

If we use `IConfigSource` in `RedisStorageProxy`, then we'll have to create an `initialize()` method in `RedisStorageProxy` that also calls the same 3 methods. This would be necessary because these methods are how the configuration properties are read into DM.

Recalling these methods is wasteful as env vars would be read repeatedly, and `configSource.initialize()` would make multiple calls to SimpleDB and would read `dynomitemanager.properties` multiple times. If we use Archaius, then multiple calls would be made to it. etc.

### Alternative solutions

I don't see any good solutions other than centralizing configuration in `DynomitemanagerConfiguration`. However, I'm not married to this solution, so if there's a better way to allow the user to customize Redis configuration, then I'm all ears.